cc_types.c

```diff
@@ -163,11 +199,204 @@ struct type* lookup_type(char* s, struct type* start)
 
 struct type* lookup_primitive_type(void)
 {
+	if(BOOTSTRAP_MODE)
+	{
+		return lookup_type(global_token->s, prim_types);
+	}
+
+	/* Lookup order for multi token types
+
+	 * unsigned
+     *	 char
+     *	 short
+     *	 short int
+     *	 long
+     *	 long int
+     *	 long long
+     *	 long long int
+     *	 int
+     *	 - (unsigned int)
+
+	 * signed
+     *	 char
+     *	 short
+     *	 short int
+     *	 long
+     *	 long int
+     *	 long long
+     *	 long long int
+     *	 int
+	 *	 - (int)
+
+	 * short
+     *	 int
+     *	 - (short)
+
+	 * long
+     *	 int
+     *	 long
+     *	 long long int
+     *	 - (long)
+	 */
+
+	if(match("unsigned", global_token->s))
+	{
+		require(global_token->next != NULL, "NULL token received in multi token type lookup");
+
+		if(match("char", global_token->next->s))
+		{
+			global_token = global_token->next;
+			return unsigned_char;
+		}
+		else if(match("short", global_token->next->s))
+		{
+			global_token = global_token->next;
+			require(global_token->next != NULL, "NULL token received in multi token type lookup 'unsigned short'");
+
+			if(match("int", global_token->next->s))
+			{
+				global_token = global_token->next;
+				/* fallthrough to unsigned_short */
+			}
+
+			return unsigned_short;
+		}
+		else if(match("long", global_token->next->s))
+		{
+			global_token = global_token->next;
+			require(global_token->next != NULL, "NULL token received in multi token type lookup 'unsigned long'");
+
+			if(match("long", global_token->next->s))
+			{
+				global_token = global_token->next;
+				require(global_token->next != NULL, "NULL token received in multi token type lookup 'unsigned long long'");
+
+				if(match("int", global_token->next->s))
+				{
+					global_token = global_token->next;
+					/* fallthrough to unsigned_long_long */
+				}
+
+				return unsigned_long_long;
+			}
+			else if(match("int", global_token->next->s))
+			{
+				global_token = global_token->next;
+				/* fallthrough to unsigned_long */
+			}
+
+			return unsigned_long;
+		}
+		else if(match("int", global_token->next->s))
+		{
+			global_token = global_token->next;
+			/* fallthrough to unsigned_integer */
+		}
+
+		return unsigned_integer;
+	}
+	else if(match("signed", global_token->s))
+	{
+		require(global_token->next != NULL, "NULL token received in multi token type lookup");
+
+		if(match("char", global_token->next->s))
+		{
+			global_token = global_token->next;
+			return signed_char;
+		}
+		else if(match("short", global_token->next->s))
+		{
+			global_token = global_token->next;
+			require(global_token->next != NULL, "NULL token received in multi token type lookup 'unsigned short'");
```

Comment:
there are still more `unsigned's`...

Reply:
Oops. Was only looking at the code and didn't see the comments since they were indented so much. :/